src/tcp_socket.cpp

```diff
@@ -300,11 +300,14 @@ namespace ip {
 		boost::system::error_code ec;
 		if (m_bound_to.address() == ip::address())
```

Comment:
so this matches both ``address_v4::any()`` and ``address_v6::any()``. That's good. It may warrant a comment